codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -77,29 +147,30 @@ public void onStreamClosed(Http2Stream stream) {
             }
 
             @Override
-            public void onPriorityTreeParentChanged(Http2Stream stream, Http2Stream oldParent) {
-                Http2Stream parent = stream.parent();
-                if (parent != null) {
-                    State state = state(stream);
-                    if (state.activeCountForTree != 0) {
-                        State pState = state(parent);
-                        pState.offerAndInitializePseudoTime(state);
-                        pState.activeCountChangeForTree(state.activeCountForTree);
-                    }
+            public void onStreamRemoved(Http2Stream stream) {
+                // The stream has been removed from the stream map. We can no longer rely on the stream's property
+                // storage to track the State. If we have room, and the precedence of the stream is sufficient, we
+                // should retain the State in the stateOnlyMap.
+                State state = state(stream);
+                state.stream = null; // If the stream was not activated, it may not be closed (reserved streams).
+                if (WeightedFairQueueByteDistributor.this.maxStateOnlySize == 0) {
```

Comment:
I'm sure its possible ... but what is the motivation and do you have a suggestion :).

The code is the way it is now to avoid using an empty queue.